pkg/safebuffer/safebuffer.go

```diff
@@ -0,0 +1,30 @@
+package safebuffer
+
+// from https://gist.github.com/arkan/5924e155dbb4254b64614069ba0afd81
+
+import (
+	"bytes"
+	"sync"
+)
+
+// Buffer is a goroutine safe bytes.Buffer
+type Buffer struct {
+	buffer bytes.Buffer
+	mutex  sync.Mutex
+}
```

Comment:
Having to introduce this is kind of a smell: You can have multiples goroutines writing concurently to your buffer, which in some cases can be OK but I suspect it is not in yours :smiley_cat:.